### PR TITLE
[RFC] "halfway a line" is a very confusing phrase

### PR DESCRIPTION
**Pull request by JackDanger**
If you Google for this phrase found in the Vim documentation you'll find
almost exclusively hits from the Vim documentation. I think changing
"halfway a line" to "halfway through a line" makes more sense.

### CODE REVIEW

**Comment by Pyrohh** [comment]
Good catch, although `git grep -i 'halfway a'` shows there are 29 more instances throughout the project.

**Comment by JackDanger** [comment]
I started with just 'halfway a line' but now I'm on a mission to fix those others too.

**Comment by Pyrohh** [comment]
:+1:

**Comment by JackDanger** [comment]
I've updated 92 really oddball uses of the word 'halfway'. Mostly turned them into 'halfway through' but sometimes just rewrote the docs for clarity.

**Comment by Pyrohh** [comment]
All done, thanks for waiting.

File: runtime/doc/todo.txt [hunk 1]
```diff
@@ -2784,7 +2784,7 @@ Most interesting new features to be added when all bugs have been fixed:
     Especially when using the scrollbar.  Typing a cursor-movement command
     scrolls back to where the cursor is.
 -   Scroll commands by screen line. g CTRL-E and g CTRL-Y ?  Requires the
-    first line to be able to start halfway.
+    first line to be able to start halfway through.
```
Comment (justinmk) [inline runtime/doc/todo.txt]: I think this is over-eager

**Comment by justinmk** [comment]
Hmm, isn't "halfway" kind of a Vim idiom? The fact that it touches so many lines also makes this seem a bit too drastic for dubious gain.

File: runtime/doc/todo.txt (hunk 1 above)
Comment (JackDanger) [inline runtime/doc/todo.txt]: This one confused me. I honestly just have no idea what the original intention was so I guessed.

**Comment by Pyrohh** [comment]
@justinmk It seems that way, but it still sounds awkward. I might agree that the benefits created by this PR would be questionable if it were only that idiom, but there are plenty of other changes done which IMO are much better. You have to start somewhere, although I assume you're talking about this introducing difficulties for merging patches.

**Comment by JackDanger** [comment]
> Hmm, isn't "halfway" kind of a Vim idiom?

If it is it's one that's hostile to any newcomer. I'm suggesting these changes because I don't want the next person to be as confused as I was.

**Comment by JackDanger** [comment]
@justinmk What do you think? Worth keeping or worth changing?

**Comment by justinmk** [comment]
@JackDanger I'm in favor of this if you remove any of the "questionable" cases, i.e. any cases where the previous wording is arguably as readable (or more readable) as the new wording.

File: runtime/doc/autocmd.txt [hunk 2]
```diff
@@ -864,9 +864,9 @@ TermResponse			After the response to |t_RV| is received from
 				the terminal.  The value of |v:termresponse|
 				can be used to do things depending on the
 				terminal version.  Note that this event may be
-				triggered halfway executing another event,
-				especially if file I/O, a shell command or
-				anything else that takes time is involved.
+				triggered halfway through another event
```
Comment (justinmk) [inline runtime/doc/autocmd.txt]: questionable

File: runtime/doc/motion.txt [hunk 3]
```diff
@@ -33,7 +33,8 @@ know what the hjkl keys do.  The mnemonic value of hjkl is clear from looking
 at the keyboard.  Think of j as an arrow pointing downwards.
 
 The 'virtualedit' option can be set to make it possible to move the cursor to
-positions where there is no character or halfway a character.
+positions where there is no character or into the middle of a multi-column
```
Comment (justinmk) [inline runtime/doc/motion.txt]: `within a multi-column`

File: runtime/doc/pattern.txt [hunk 4]
```diff
@@ -938,7 +938,7 @@ $	At end of pattern or in front of "\|", "\)" or "\n" ('magic' on):
 	values of the current window are used (e.g., 'tabstop').
 	The "23" can be any column number.  The first column is 1.
 	Note that some virtual column positions will never match, because they
-	are halfway through a tab or other character that occupies more than
+	are in the middle of a tab or other character that occupies more than
```
Comment (justinmk) [inline runtime/doc/pattern.txt]: seems questionable

File: runtime/doc/syntax.txt [hunk 5]
```diff
@@ -4075,8 +4075,8 @@ the match doesn't move to another line.
 The skip pattern can include the "\n", but the search for an end pattern will
 continue in the first character of the next line, also when that character is
 matched by the skip pattern.  This is because redrawing may start in any line
-halfway a region and there is no check if the skip pattern started in a
-previous line.	For example, if the skip pattern is "a\nb" and an end pattern
+halfway through a region and there is no check if the skip pattern started in
```
Comment (justinmk) [inline runtime/doc/syntax.txt]: `halfway in` ?

File: runtime/doc/usr_03.txt [hunk 6]
```diff
@@ -200,7 +200,8 @@ more typing.
 		last line of a file    V
 
 Another way to move to a line is using the "%" command with a count.  For
-example "50%" moves you to halfway the file.  "90%" goes to near the end.
+example, "50%" moves you to halfway through the file, and "90%" goes to near
```
Comment (justinmk) [inline runtime/doc/usr_03.txt]: moves you ~~to~~ halfway through the file

File: runtime/doc/usr_04.txt [hunk 7]
```diff
@@ -201,7 +201,8 @@ change.  Then you can use Visual mode.
 You start Visual mode by pressing "v".  You move the cursor over the text you
 want to work on.  While you do this, the text is highlighted.  Finally type
 the operator command.
-   For example, to delete from halfway one word to halfway another word:
+   For example, to delete from halfway through one word to halfway through
+   another word:
```
Comment (justinmk) [inline runtime/doc/usr_04.txt]: new version is just too wordy

File: runtime/doc/usr_05.txt [hunk 8]
```diff
@@ -133,7 +133,7 @@ the "2fw" command is executed and the displayed "2f" is removed.
 >
 	set incsearch
 
-Display the match for a search pattern when halfway typing it.
+Display matches for a search pattern while you type.
```
Comment (justinmk) [inline runtime/doc/usr_05.txt]: :+1:

File: runtime/doc/usr_07.txt [hunk 9]
```diff
@@ -233,8 +233,8 @@ another file and place marks there, these are specific for that file.  Thus
 each file has its own set of marks, they are local to the file.
    So far we were using marks with a lowercase letter.  There are also marks
 with an uppercase letter.  These are global, they can be used from any file.
-For example suppose that we are editing the file "foo.txt".  Go to halfway the
-file ("50%") and place the F mark there (F for foo): >
+For example suppose that we are editing the file "foo.txt".  Go to halfway
+through the file ("50%") and place the F mark there (F for foo): >
```
Comment (justinmk) [inline runtime/doc/usr_07.txt]: "of" might be better

File: runtime/doc/usr_07.txt [hunk 10]
```diff
@@ -242,8 +242,8 @@ Now edit the file "bar.txt" and place the B mark (B for bar) at its last line:
 >
 	GmB
 
-Now you can use the "'F" command to jump back to halfway foo.txt.  Or edit yet
-another file, type "'B" and you are at the end of bar.txt again.
+Now you can use the "'F" command to jump back to halfway through foo.txt.  To
```
Comment (justinmk) [inline runtime/doc/usr_07.txt]: of

File: runtime/doc/usr_07.txt [hunk 11]
```diff
@@ -242,8 +242,8 @@ Now edit the file "bar.txt" and place the B mark (B for bar) at its last line:
...
+edit the other file, type "'B" and you jump to the end of bar.txt.
```
Comment (justinmk) [inline runtime/doc/usr_07.txt]: "To edit the other file" isn't the intent of the old sentence

File: runtime/doc/usr_10.txt [hunk 12]
```diff
@@ -370,7 +370,8 @@ into "barfoo".
 was specified in this example.  This is different from ":substitute", which
 works on one line without a range.
    The command isn't perfect, since it also matches lines where "//" appears
-halfway a line, and the substitution will also take place before the "//".
+halfway through a line, and the substitution will also take place before the
```
Comment (justinmk) [inline runtime/doc/usr_10.txt]: `halfway in`

File: runtime/doc/usr_10.txt [hunk 13]
```diff
@@ -625,7 +626,7 @@ When you are typing plain text, it's nice if the length of each line is
 automatically trimmed to fit in the window.  To make this happen while
 inserting text, set the 'textwidth' option: >
 
-	:set textwidth=72
+	:set textwidth=78
```
Comment (justinmk) [inline runtime/doc/usr_10.txt]: haha, what?

File: runtime/doc/usr_21.txt [hunk 14]
```diff
@@ -155,11 +155,11 @@ to be lost.  Each item can be remembered only once.
 
 GETTING BACK TO WHERE YOU STOPPED VIM
 
-You are halfway editing a file and it's time to leave for holidays.  You exit
-Vim and go enjoy yourselves, forgetting all about your work.  After a couple
-of weeks you start Vim, and type:
+You are halfway through editing a file and it's time to leave for holidays.
```
Comment (justinmk) [inline runtime/doc/usr_21.txt]: questionable

File: runtime/doc/usr_23.txt [hunk 15]
```diff
@@ -61,10 +61,10 @@ The three names that Vim uses are:
 USING THE MAC FORMAT
 
 On Unix, <LF> is used to break a line.  It's not unusual to have a <CR>
-character halfway a line.  Incidentally, this happens quite often in Vi (and
-Vim) scripts.
+character halfway through a line.  Incidentally, this happens quite often in
```
Comment (justinmk) [inline runtime/doc/usr_23.txt]: halfway in

File: runtime/doc/usr_23.txt [hunk 16]
```diff
@@ -61,10 +61,10 @@ The three names that Vim uses are:
...
+Vi (and Vim) scripts.
    On the Macintosh, where <CR> is the line break character, it's possible to
-have a <LF> character halfway a line.
+have a <LF> character halfway through a line.
```
Comment (justinmk) [inline runtime/doc/usr_23.txt]: halfway in

File: runtime/doc/usr_25.txt [hunk 17]
```diff
@@ -354,8 +354,8 @@ BREAKING AT WORDS				*edit-no-break*
 
 When preparing text for use by another program, you might have to make
 paragraphs without a line break.  A disadvantage of using 'nowrap' is that you
-can't see the whole sentence you are working on.  When 'wrap' is on, words are
-broken halfway, which makes them hard to read.
+can't see the whole sentence you are working on.  When 'wrap' is on, words can
+begin on one line and end on the next, making them hard to read.
```
Comment (justinmk) [inline runtime/doc/usr_25.txt]: I don't think this is worth changing. The old wording was fine

File: runtime/doc/usr_29.txt [hunk 18]
```diff
@@ -223,8 +223,8 @@ example, suppose you want to select a tag that starts with "write_": >
 	:tselect /^write_
 
 The "^" specifies that the tag starts with "write_".  Otherwise it would also
-be found halfway a tag name.  Similarly "$" at the end makes sure the pattern
-matches until the end of a tag.
+be found halfway through a tag name.  Similarly "$" at the end makes sure the
```
Comment (justinmk) [inline runtime/doc/usr_29.txt]: halfway in

File: runtime/doc/usr_40.txt [hunk 19]
```diff
@@ -172,8 +172,9 @@ thus doing the substitution, etc.
 where the substitute command doesn't find a match for "5.1".  You can then
 make a change to insert "5.1" and continue by typing ",," again.  Or the
 ":wnext" fails, because you are in the last file in the list.
-   When a mapping runs into an error halfway, the rest of the mapping is
-discarded.  CTRL-C interrupts the mapping (CTRL-Break on MS-Windows).
+   A mapping executes until it finishes or encounters an error, where an error
+causes the rest of the mapping to be discarded.  CTRL-C interrupts the
+mapping (CTRL-Break on MS-Windows).
```
Comment (justinmk) [inline runtime/doc/usr_40.txt]: old wording was fine, and more brief

File: runtime/doc/various.txt [hunk 20]
```diff
@@ -19,9 +19,9 @@ CTRL-L			Clear and redraw the screen.  The redraw may happen
 							*:redr* *:redraw*
 :redr[aw][!]		Redraw the screen right now.  When ! is included it is
 			cleared first.
-			Useful to update the screen halfway executing a script
-			or function.  Also when halfway a mapping and
-			'lazyredraw' is set.
+			Useful to update the screen halfway through executing
+			a script or function.  Also when halfway through a
+			mapping and 'lazyredraw' is set.
```
Comment (justinmk) [inline runtime/doc/various.txt]: > Useful to update the screen during execution of a script or function (or a mapping, if 'lazyredraw' is set).

File: runtime/doc/visual.txt [hunk 21]
```diff
@@ -492,7 +492,8 @@ Otherwise, typed characters are handled as in Visual mode.
 
 When using an operator in Select mode, and the selection is linewise, the
 selected lines are operated upon, but like in characterwise selection.  For
-example, when a whole line is deleted, it can later be pasted halfway a line.
+example, when a whole line is deleted, it can later be pasted halfway through
+a line.
```
Comment (justinmk) [inline runtime/doc/visual.txt]: "in the middle of a line" ?

File: runtime/syntax/c.vim [hunk 22]
```diff
@@ -204,8 +204,8 @@ if exists("c_comment_strings")
   syn region cComment2String	contained start=+L\=\\\@<!"+ skip=+\\\\\|\\"+ end=+"+ end="$" contains=cSpecial
   syn region  cCommentL	start="//" skip="\\$" end="$" keepend contains=@cCommentGroup,cComment2String,cCharacter,cNumbersCom,cSpaceError,@Spell
   if exists("c_no_comment_fold")
-    " Use "extend" here to have preprocessor lines not terminate halfway a
-    " comment.
+    " Use "extend" here to have preprocessor lines not terminate halfway
+    " through a comment.
```
Comment (justinmk) [inline runtime/syntax/c.vim]: We want to avoid modifying most `runtime/` files becaues it makes merges very easy, and we have no interest in maintaining them for the time being.

File: src/nvim/edit.c [hunk 23]
```diff
@@ -766,7 +766,7 @@ edit (
 #ifdef UNIX
 do_intr:
 #endif
-      /* when 'insertmode' set, and not halfway a mapping, don't leave
+      /* when 'insertmode' set, and not halfway through a mapping, don't leave
        * Insert mode */
```
Comment (justinmk) [inline src/nvim/edit.c]: might as well change to `//` comment style.

File: src/nvim/hardcopy.c [hunk 24]
```diff
@@ -827,7 +827,7 @@ static colnr_T hardcopy_line(prt_settings_T *psettings, int page_line, prt_pos_T
       prt_line_number(psettings, page_line, ppos->file_line);
     ppos->ff = FALSE;
   } else {
-    /* left over from wrap halfway a tab */
+    /* left over from wrap halfway through a tab */
```
Comment (justinmk) [inline src/nvim/hardcopy.c]: `//`

File: src/nvim/move.c [hunk 25]
```diff
@@ -1630,8 +1630,8 @@ void scroll_cursor_bot(int min_scroll, int set_topbot)
 }
 
 /*
- * Recompute topline to put the cursor halfway the window
- * If "atend" is TRUE, also put it halfway at the end of the file.
+ * Recompute topline to put the cursor halfway across the window
+ * If "atend" is TRUE, also put it halfway to the end of the file.
  */
```
Comment (justinmk) [inline src/nvim/move.c]: `///` (triple slash, for javadoc)

File: runtime/doc/usr_10.txt (hunk 13 above)
Comment (Pyrohh) [inline runtime/doc/usr_10.txt]: The paragraph below references that value being set from the example vimrc, but the example vimrc sets `78`.

Comment (justinmk) [inline runtime/doc/usr_10.txt]: ok

File: runtime/doc/autocmd.txt (hunk 2 above)
Comment (JackDanger) [inline runtime/doc/autocmd.txt]: How about:
`triggered halfway executing another event`
->
`triggered while executing another event`
?

File: runtime/doc/motion.txt (hunk 3 above)
Comment (JackDanger) [inline runtime/doc/motion.txt]: ackknowledge.

File: runtime/doc/pattern.txt (hunk 4 above)
Comment (JackDanger) [inline runtime/doc/pattern.txt]: agreed, reverting.

File: runtime/doc/syntax.txt (hunk 5 above)
Comment (JackDanger) [inline runtime/doc/syntax.txt]: great. changing.

File: runtime/doc/usr_03.txt (hunk 6 above)
Comment (JackDanger) [inline runtime/doc/usr_03.txt]: changed.

File: runtime/doc/usr_04.txt (hunk 7 above)
Comment (JackDanger) [inline runtime/doc/usr_04.txt]: Changing:
`to delete from halfway through one word to halfway through another word:`
->
`to delete from the middle of a word to the middle of another:` (fits on one line)

File: runtime/doc/usr_07.txt (hunk 9 above)
Comment (JackDanger) [inline runtime/doc/usr_07.txt]: changed.

File: runtime/doc/usr_07.txt (hunk 10 above)
Comment (JackDanger) [inline runtime/doc/usr_07.txt]: 'to halfway of foo.txt' ? I wouldn't ever use 'of' in like that but I'll change if it you prefer it.

File: runtime/doc/usr_07.txt (hunk 11 above)
Comment (JackDanger) [inline runtime/doc/usr_07.txt]: I missed that - great catch.

File: runtime/doc/usr_10.txt (hunk 12 above)
Comment (JackDanger) [inline runtime/doc/usr_10.txt]: changed.

File: runtime/doc/usr_21.txt (hunk 14 above)
Comment (JackDanger) [inline runtime/doc/usr_21.txt]: I disagree on this one - the original is non-parseable English. "To be halfway editing a file" sounds like you're half editing and half cooking dinner or half editing and half asleep.

File: runtime/doc/usr_23.txt (hunk 15 above)
Comment (JackDanger) [inline runtime/doc/usr_23.txt]: changed.

File: runtime/doc/usr_23.txt (hunk 16 above)
Comment (JackDanger) [inline runtime/doc/usr_23.txt]: changed.

File: runtime/doc/usr_25.txt (hunk 17 above)
Comment (JackDanger) [inline runtime/doc/usr_25.txt]: You're totally right, reverting.

File: runtime/doc/usr_29.txt (hunk 18 above)
Comment (JackDanger) [inline runtime/doc/usr_29.txt]: changed.

File: runtime/doc/usr_40.txt (hunk 19 above)
Comment (JackDanger) [inline runtime/doc/usr_40.txt]: Please see prior discussion with @Pyrohh here: https://github.com/neovim/neovim/commit/54a61ef98159de1b24a584661b5659ee7ddd111a#commitcomment-9351867

File: runtime/doc/various.txt (hunk 20 above)
Comment (JackDanger) [inline runtime/doc/various.txt]: changed.

File: runtime/doc/visual.txt (hunk 21 above)
Comment (JackDanger) [inline runtime/doc/visual.txt]: great.

File: runtime/syntax/c.vim (hunk 22 above)
Comment (JackDanger) [inline runtime/syntax/c.vim]: Sounds smart. Reverting.

File: src/nvim/edit.c (hunk 23 above)
Comment (JackDanger) [inline src/nvim/edit.c]: changed.

File: src/nvim/hardcopy.c (hunk 24 above)
Comment (JackDanger) [inline src/nvim/hardcopy.c]: changed.

File: src/nvim/move.c (hunk 25 above)
Comment (JackDanger) [inline src/nvim/move.c]: Great. I just learned we're doing this today and I'm stoked to make these doc updates.

Comment (JackDanger) [inline src/nvim/move.c]: I've updated the comment style but I'm having a really hard time understanding what the `atend` parameter does exactly so I didn't try rephrasing its description.

File: runtime/doc/usr_40.txt (hunk 19 above)
Comment (justinmk) [inline runtime/doc/usr_40.txt]: Noted, but I just don't think this is worth changing. I can't do it, captain!

Comment (JackDanger) [inline runtime/doc/usr_40.txt]: Fair enough :) Reverting.

File: runtime/doc/various.txt [hunk 26]
```diff
@@ -19,9 +19,9 @@ CTRL-L			Clear and redraw the screen.  The redraw may happen
 							*:redr* *:redraw*
 :redr[aw][!]		Redraw the screen right now.  When ! is included it is
 			cleared first.
-			Useful to update the screen halfway executing a script
-			or function.  Also when halfway a mapping and
-			'lazyredraw' is set.
+                        Useful to update the screen halfway through executing
+                        a script or function (or a mapping if 'lazyredraw'
+                        set).
```
Comment (justinmk) [inline runtime/doc/various.txt]: Sadly, we must preserve tabs in the help files. Otherwise we will have a never-ending formatting mess. Can you change this back to tabs and make sure the other changes are as well?

Comment (justinmk) [inline runtime/doc/various.txt]: https://github.com/tpope/vim-sleuth might help with this, as we as `set list`.

**Comment by justinmk** [comment]
LGTM after restoring the tab indentation

**Comment by JackDanger** [comment]
> LGTM after restoring the tab indentation

This is my bad, my editor settings have aggressive defaults :( Fixing.

**Comment by JackDanger** [comment]
> This is my bad, my editor settings have aggressive defaults :( Fixing.

And by "defaults" I mean "settings I imposed to find and slaughter all tabs" ¯\_(ツ)_/¯

**Comment by JackDanger** [comment]
Okay, I believe I've implemented all feedback. Thanks for the help in the reviews!
